routes/distribution.js

```diff
@@ -6,67 +6,51 @@ import {
   missingHandler,
 } from './helperMethods.js';
 import { auth } from '../util/token.js';
-import Courses from '../model/Course.js';
 import Distributions from '../model/Distribution.js';
 import Plans from '../model/Plan.js';
+import Majors from '../model/Major.js';
 import express from 'express';
+import { initDistributions } from './distributionMethods.js';
 
 const router = express.Router();
 
 //get distribution by id
 router.get('/api/distributions/:distribution_id', auth, async (req, res) => {
   const d_id = req.params.distribution_id;
   try {
-    const distribution = await Distributions.findById(d_id).exec();
-    // verify that distribution belongs to user
-    if (req.user._id !== distribution.user_id) {
-      forbiddenHandler(res);
-    } else {
-      returnData(distribution, res);
-    }
+    const distribution = await Distributions.findById(d_id)
+      .populate('fineReq_ids')
+      .exec();
+    if (!distribution)
+      return errorHandler(res, 404, { message: 'Distribution not found.' });
+    returnData(distribution, res);
   } catch (err) {
-    errorHandler(res, 400, err);
+    errorHandler(res, 500, err);
   }
 });
 
 //get all distributions in a plan
-router.get('/api/distributionsByPlan/:plan_id', auth, async (req, res) => {
-  const plan_id = req.params.plan_id;
+router.get('/api/distributionsByPlan/', async (req, res) => {
+  const plan_id = req.query.plan_id;
+  const major_id = req.query.major_id;
+  if (!plan_id || !major_id) return missingHandler(res, { plan_id, major_id });
+  const reload = req.query.reload;
```

Comment:
returns all existing distributions for specified plan / major 

if reload === 'true', re initializes distributions